### PR TITLE
Adding first proxy version

### PR DESCRIPTION
**Pull request by j-clap**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,160 @@
...
+## Usage
+The PostgreSQL adapter can be started both as a standalone process as well as an 
+in-process server.
+
+### Standalone
+1. Build a jar file containing all dependencies by running `mvn package`.
+2. Execute `java -jar <jar-file> <options>`.
+
+The following options are required to run the proxy:
+* p
```
Comment (bvandiver) [inline README.md]: formatting nit: fixed width font and perhaps `-p <projectname>`
I found that the single characters got lost / looked a little odd next to the bullets.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,160 @@
...
+  * The Spanner database name.
+* c
+  * The full path for the file containing the service account credentials in
+    JSON format.
+  * Instructions for generating the aforementioned file are found [here](https://cloud.google.com/iam/docs/creating-managing-service-account-keys)
+  * Do remember to give the service account sufficient credentials to access the
+    database.
+
+The following options are optional:
+* s
```
Comment (bvandiver) [inline README.md]: ditto for these

File: README.md [hunk 3]
```diff
@@ -1,2 +1,160 @@
-# pgadapter
-A proxy that translates the wire protocol from Postgres to Cloud Spanner.
+# Google Cloud Spanner PostgreSQL Adapter
+
+## Introduction
+Spanner-PGAdapter is a simple, MITM, forward, non-transparent proxy, which 
+translates Postgres wire protocol into the Cloud Spanner equivalent. By running
+this proxy locally, any Postgres client (including PSQL) should function
```
Comment (bvandiver) [inline README.md]: elaborate here: including the SQL command-line client `psql`

File: README.md [hunk 4]
```diff
@@ -1,2 +1,160 @@
...
+    know what extended query mode and simple query mode is, then you should 
+    probably not be using this setting.
+* j
+  * The full path for a file containing a JSON object to do SQL translation
+  based on RegEx replacement. Any item matching the input_pattern will be 
+  replaced by the output_pattern string, wherein capture groups are allowed
+  and their order is specified via the matcher_array item. Match replacement
+  must be defined via %s in the output_pattern string. Set matcher_array to []
+  if no matches exist. User-specified patterns will precede internal matches.
+  Escaped and general regex syntax matches Java RegEx syntax.
```
Comment (bvandiver) [inline README.md]: Can we link this to a page on the regex syntax?

**Review by bvandiver — APPROVED** [review]
Some nits for the readme.
I think we still need to untangle some stuff in translation, but I think we can do that after the initial version goes out.

File: README.md (hunk 1 above)
Comment (j-clap) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (j-clap) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (j-clap) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (j-clap) [reply README.md]: Done

**Pull request merged**